phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1530,8 +1533,8 @@ private void ensureViewIndexTableCreated(byte[] physicalTableName, Map<String, O
         byte[] physicalIndexName = MetaDataUtil.getViewIndexPhysicalName(physicalTableName);
 
         tableProps.put(MetaDataUtil.IS_VIEW_INDEX_TABLE_PROP_NAME, TRUE_BYTES_AS_STRING);
-        TableDescriptor desc = ensureTableCreated(physicalIndexName, PTableType.TABLE, tableProps, families, splits,
-                false, isNamespaceMapped, false);
+        TableDescriptor desc = ensureTableCreated(physicalIndexName, PTableType.INDEX, tableProps, families, splits,
+                true, isNamespaceMapped, false);
```

Comment:
@kadirozde This is fixing the problem I encountered. The fix makes sense to me. I ll create another jira for this.

Reply:
@kadirozde for your review https://github.com/apache/phoenix/pull/531

Reply:
Got it